ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala

```diff
@@ -177,4 +186,95 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
     }
   })
 
+  test(
+    "DuplicateSubmissionIdTwiceCorrect",
+    "Changing config twice with the same submissionId succeeds only on two participants",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    // Multiple config changes should never cause duplicates. Participant adds extra entropy to the
+    // submission id to ensure client does not inadvertently cause problems by poor selection
+    // of submission ids.
+    for {
+      (req1, req2) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 1)
+      )
+      _ <- ignoreNotAuthorized(alpha.setTimeModel(req1))
+      _ <- ignoreNotAuthorized(beta.setTimeModel(req2))
+    } yield ()
+
+  })
+
+  test(
+    "DuplicateSubmissionIdFirstIncorrect",
+    "Changing config twice with the same submissionId fails for first incorrect submission",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    for {
+      (good, bad) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 2)
+      )
+      failure <- ignoreNotAuthorized(beta.setTimeModel(bad))
+        .mustFail("Mismatching configuration generation")
+      _ <- alpha.setTimeModel(good)
+    } yield {
+      assertGrpcError(failure, Status.Code.INVALID_ARGUMENT, "Mismatching configuration generation")
+    }
+  })
+
+  test(
+    "DuplicateSubmissionIdSecondIncorrect",
+    "Changing config twice with the same submissionId fails for first incorrect submission",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    for {
+      (good, bad) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 2)
+      )
+      _ <- alpha.setTimeModel(good)
+      failure <- ignoreNotAuthorized(beta.setTimeModel(bad))
+        .mustFail("Mismatching configuration generation")
+    } yield {
+      assertGrpcError(failure, Status.Code.INVALID_ARGUMENT, "Mismatching configuration generation")
+    }
+  })
+
+  def generateRequest(
+      participant: ParticipantTestContext
+  )(implicit ec: ExecutionContext): Future[SetTimeModelRequest] =
+    for {
+      response <- participant.getTimeModel()
+      oldTimeModel = {
+        assert(response.timeModel.isDefined, "Expected time model to be defined")
+        response.timeModel.get
```

Comment:
A tad shorter and a bit more idiomatic
```suggestion
      oldTimeModel = response.timeModel.getOrElse(
        throw new AssertionError("Expected time model to be defined")
      )
```

Reply:
Done